library/network/test/y2firewall/firewalld_test.rb

```diff
@@ -179,4 +181,142 @@
       expect(firewalld.api).to be_a Y2Firewall::Firewalld::Api
     end
   end
+
+  describe "#read" do
+    let(:zones_definition) do
+      ["dmz",
+       "  target: default",
+       "  interfaces: ",
+       "  ports: ",
+       "  protocols:",
+       "",
+       "external (active)",
+       "  target: default",
+       "  interfaces: eth0",
+       "  services: ssh samba",
+       "  ports: 5901/tcp 5901/udp",
+       "  protocols: "]
+    end
+
+    let(:api) do
+      instance_double(Y2Firewall::Firewalld::Api,
+        log_denied_packets: false,
+        default_zone:       "dmz",
+        list_all_zones:     zones_definition,
+        zones:              known_zones)
+    end
+
+    before do
+      allow(firewalld).to receive("api").and_return api
+    end
+
+    it "returns false if firewalld is not installed" do
+      allow(firewalld).to receive(:installed?).and_return(false)
+
+      expect(firewalld.read).to eq(false)
+    end
+
+    it "initializes @zones parsing the firewalld zones list" do
```

Comment:
I would replace *@zones* by something like *the list of zones* (being a instance variable is an implementation detail).